AsyncDisplayKit/Details/ASRangeController.mm

```diff
@@ -79,63 +33,32 @@ @implementation ASRangeController
 - (instancetype)init {
   if (self = [super init]) {
 
-    _workingRangeIndexPaths = [NSSet set];
-    _workingRangeIsValid = YES;
+    _rangeIsValid = YES;
+    _rangeTypeIndexPaths = [[NSMutableDictionary alloc] init];
+
+    _rangeTypeHandlers = @{
+                            @(ASLayoutRangeTypeRender): [[ASRangeHandlerRender alloc] init],
+                            @(ASLayoutRangeTypePreload): [[ASRangeHandlerPreload alloc] init],
```

Comment:
Not related.
I like the ASRangeHandlerXX, and it may could be refactored it from RangeController using `strategy` pattern.
And the RangeController doesn't need to know about Specific RangeHandler.